heareval/tasks/pipeline.py

```diff
@@ -550,12 +652,18 @@ def requires(self):
         }
 
     def run(self):
-        for audiofile in tqdm(list(self.requires()["corpus"].workdir.glob("*.wav"))):
-            # Compare the filename with the slug.
-            # Note that the slug does not have the extension of the file
-            split = self.metadata.loc[
-                self.metadata["slug"] == audiofile.stem, "split"
-            ].values[0]
+        audiofiles = list(self.requires()["corpus"].workdir.glob("*.wav"))
+        assert len(audiofiles) == len(
+            self.metadata["unique_filestem"].drop_duplicates()
+        )
+        for audiofile in audiofiles:
+            # Compare the filename with the unique_filestem.
+            # Note that the unique_filestem does not have a file extension
+            splits = self.metadata.loc[
+                self.metadata["unique_filestem"] == audiofile.stem, "split"
+            ].drop_duplicates()
+            assert len(splits) == 1
```

Comment:
Will this assertion also hold? I thought that for something like speech commands that the unique_filestem might include the parent folder

Reply:
Adding comment.

unique_filestem should be unique across the entire dataset and imply a particular split.

Reply:
That's its definition.

Reply:
Yes I understand -- we also use the unique_filestem for the names of the audio files down the pipeline.